Explorer/Assets/DCL/InWorldCamera/InWorldCamera/Systems/CaptureScreenshotSystem.cs

```diff
@@ -64,30 +76,46 @@ protected override void Update(float t)
 
             if (recorder.State == RecordingState.SCREENSHOT_READY && metadataBuilder.MetadataIsReady)
             {
-                // TODO (Vit): This is a temporary solution for debug purposes. Will be replaced by proper MVC + sending to backend
-                {
-                    hud.Screenshot = recorder.GetScreenshotAndReset();
-                    hud.Metadata = metadataBuilder.GetMetadataAndReset();
-
-                    cameraReelStorageService.UploadScreenshotAsync(hud.Screenshot, hud.Metadata, ctx.Token).Forget();
-
-                    hud.Canvas.enabled = true;
-                }
-
+                ProcessCapturedScreenshot();
                 return;
             }
 
-            if (recorder.State == RecordingState.IDLE && World.TryGet<InWorldCameraInput>(camera, out var input) && input.TakeScreenshot)
+            if (ScreenshotIsRequested())
             {
-                hud.Canvas.enabled = false;  // TODO (Vit): This is a temporary solution for debug puproses. Will be replaced by proper MVC
+                hudController.Hide(isInstant: true);
                 coroutineRunner.StartCoroutine(recorder.CaptureScreenshot());
                 CollectMetadata();
             }
         }
 
+        private void ProcessCapturedScreenshot()
+        {
+            screenshot = recorder.GetScreenshotAndReset();
+            metadata = metadataBuilder.GetMetadataAndReset();
+
+            cameraReelStorageService.UploadScreenshotAsync(screenshot, metadata, ctx.Token).Forget();
+
+            hudController.Show();
+            hudController.PlayScreenshotFX(screenshot, SPLASH_FX_DURATION, MIDDLE_PAUSE_FX_DURATION, IMAGE_TRANSITION_FX_DURATION);
+            hudController.DebugCapture(screenshot, metadata);
+        }
+
+        private bool ScreenshotIsRequested()
```

Comment:
Just a small suggestion, when naming methods/properties that return boolean, writing the Is-, Can- Has- as prefix helps readability.

Reply:
Sorry, but I would disagree here. I am more of following  _"Clean code reads like well-written prose"_ as said by Robert C. Martin (Uncle Bob), or Donald Knuth concept of _literate programming_. So for me `"if screenshot is requested"` is more readable than `"if is screenshot requested"`

Reply:
It's ok, I normally follow advices from "Framework Design Guidelines" by Krzysztof Cwalina and Brad Abrams. Since there is no rule in our code style guidelines you can leave it like that.